tools/internal_ci/linux/grpc_bazel_on_foundry_base.sh

```diff
@@ -57,6 +57,8 @@ source tools/internal_ci/helper_scripts/prepare_build_linux_rc
 
 if [ "$UPLOAD_TEST_RESULTS" != "" ]
 then
+  # Sleep to let ResultStore finish writing results before querying
```

Comment:
is there a risk of writing partial test results if result store takes longer or is this binary (result present / results not present -> error)?

Reply:
I think it's binary. IIRC, the error stacktrace originates from the first call to ResultStore's API, so I assume it didn't partially upload, but I didn't check. 

I don't think we're at risk of having results being partially uploaded without the entire build failing.